misc/observerability-symbol-collector/src/main/java/org/ballerinalang/observability/anaylze/DefaultObservabilitySymbolCollector.java

```diff
@@ -150,7 +150,9 @@ private PkgASTHolder getModuleASTHolder(BLangPackage module) {
         List<BLangCompilationUnit> compilationUnits = module.getCompilationUnits();
         for (BLangCompilationUnit cUnit : compilationUnits) {
             CUnitASTHolder jsonCUnit = getCUnitASTHolder(visibleEPsByNode, cUnit);
-            pkgASTHolder.addCompilationUnit(cUnit.name, jsonCUnit);
+            if (jsonCUnit.getAst() != null) {
```

Comment:
I am bit skeptical  on this null check. When can `jsonCUnit.getAst()` be null?

Reply:
When an exception is thrown while generating the AST JSON (while calling `TextDocumentFormatUtil.generateJSON`), jsonCUnit.getAst() can be null. We observed this in several scenarios. The warning in https://github.com/ballerina-platform/ballerina-lang/blob/master/misc/observerability-symbol-collector/src/main/java/org/ballerinalang/observability/anaylze/DefaultObservabilitySymbolCollector.java#L181 gets printed and AST is null.

Please refer  #24341

Reply:
Ideally Formatter shouldn't throw an error, but this is just to be sure that this doesn't break the compilation itself.

The cases in #24341 were throwing a Stackoverflow exception due to a bug (This bug had been fixed)

Reply:
It is better to show errors than hide it. IMO what we should do is provide a way to disable AST generation.